include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -677,13 +681,21 @@ __parallel_find_or(_ExecutionPolicy&& __exec, _Brick __f, _BrickTag __brick_tag,
                     // 1. Set initial value to local atomic
                     if (__local_idx == 0)
                         __found_local.store(__init_value);
+#if _ONEDPL_SYCL2020_BARRIERS_PRESENT
```

Comment:
I suggest using our own simple function-wrapper which accepts `__item_id` and placing "#if..#else" within.